core/modules/catalog/events.ts

```diff
@@ -40,8 +40,9 @@ export const filterChangedProduct = async (filterOption, store, router) => {
     }
   }, { root: true })
   const { configuration, selectedVariant, options, product_option } = newProductVariant
+  let routeProp;
   if (config.products.setFirstVarianAsDefaultInURL && selectedVariant) {
-    const routeProp = config.seo.useUrlDispatcher ? 'params' : 'query'
+    routeProp = config.seo.useUrlDispatcher ? 'params' : 'query'
```

Comment:
Let's make something to make future developers understand why in one place we have `params : query` and below we have `query : params`

Reply:
I have checked it again and it looks like we don't need to have both (`params : query` and `query : params`).
`query : params` is enough.